drivers/serial/uart_sf32lb.c

```diff
@@ -35,13 +37,45 @@
 /* minimal BRR: INT=1, FRAC=0 (0x10) */
 #define UART_BRR_MIN 0x10U
 
+struct uart_sf32lb_data {
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_user_data_t irq_callback;
+	void *cb_data;
+#endif
+};
+
 struct uart_sf32lb_config {
 	uintptr_t base;
 	const struct pinctrl_dev_config *pcfg;
 	struct sf32lb_clock_dt_spec clock;
 	struct uart_config uart_cfg;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	void (*irq_config_func)(const struct device *dev);
+#endif
 };
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static void uart_sf32lb_isr(const struct device *dev)
+{
+	const struct uart_sf32lb_config *config = dev->config;
+	struct uart_sf32lb_data *data = dev->data;
+
+	if (data->irq_callback) {
+		data->irq_callback(dev, data->cb_data);
+	}
+
+	if (sys_test_bit(config->base + UART_ISR, USART_ISR_TXE_Pos)) {
+		sys_clear_bit(config->base + UART_ICR, USART_ISR_TXE_Pos);
+	}
+	if (sys_test_bit(config->base + UART_ISR, USART_ISR_TC_Pos)) {
+		sys_clear_bit(config->base + UART_ICR, USART_ICR_TCCF_Pos);
+	}
+	if (sys_test_bit(config->base + UART_ISR, USART_ISR_RXNE_Pos)) {
+		sys_clear_bit(config->base + UART_ICR, USART_ISR_RXNE_Pos);
+	}
```

Comment:
Wouldn't that do the same? Is it needed to check flags?
```suggestion
	sys_write32(config->base + UART_ICR, USART_ISR_TXE | USART_ICR_TCCF | USART_ISR_RXNE);
```

clear bit functions do a RW, but ICR is meant to W1C